src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs

```diff
@@ -23,65 +25,239 @@ namespace System.Text.RegularExpressions.Generator
+                    }
+                }
 
+                // If we have any (RegexType regexType, string generatedName, string runnerFactoryImplementation, Dictionary<string, string[]> requiredHelpers),
+                // those are generated implementations to be emitted. For now, assign an ID, and emit the partial method.  We also gather up all of the helpers
+                // these methods will need emitted.
+                Dictionary<string, string[]> requiredHelpers = new();
                 foreach (object? result in results)
                 {
-                    switch (result)
+                    if (result is ValueTuple<RegexType, string, Dictionary<string, string[]>> regexImpl)
                     {
-                        case Diagnostic d:
-                            context.ReportDiagnostic(d);
-                            break;
+                        regexImpl.Item1.GeneratedId = id++;
 
-                        case ValueTuple<string, ImmutableArray<Diagnostic>> t:
-                            code.Add(t.Item1);
-                            foreach (Diagnostic d in t.Item2)
+                        foreach (KeyValuePair<string, string[]> helper in regexImpl.Item3)
+                        {
+                            if (!requiredHelpers.ContainsKey(helper.Key))
                             {
-                                context.ReportDiagnostic(d);
+                                requiredHelpers.Add(helper.Key, helper.Value);
                             }
-                            break;
+                        }
+
+                        EmitRegexPartialMethod(regexImpl.Item1, writer, generatedClassName, regexImpl.Item1.GeneratedId);
+                        writer.WriteLine();
+                    }
+                }
+
+                // At this point we've emitted all the partial method definitions, but we still need to emit the actual regex-derived implementations.
+                // These are all emitted inside of our generated class.
+                // TODO https://github.com/dotnet/csharplang/issues/5529:
+                // When C# provides a mechanism for shielding generated code from the rest of the project, it should be employed
+                // here for the generated class.  At that point, the generated class wrapper can be removed, and all of the types
+                // generated inside of it (one for each regex as well as the helpers type) should be shielded.
+
+                writer.WriteLine($"namespace {GeneratedNamespace}");
+                writer.WriteLine($"{{");
+
+                // We emit usings here now that we're inside of a namespace block and are no longer emitting code into
+                // a user's partial type.  We can now rely on binding rules mapping to these usings and don't need to
+                // use global-qualified names for the rest of the implementation.
+                writer.WriteLine($"    using System;");
```

Comment:
Do we need a new pragma warnings in the headers though? I believe it is common for people to add stylecop rules that require using statements to be outside of a namespace.

Reply:
Such rules by default shouldn't apply to generated code. But even if they did, I don't want us to add pragmas for warnings that aren't in the .NET SDK; otherwise we open the doors to an unending laundry list of arbitrary 3rd-party analyzers we'd need to start considering.

Note there is a hidden warning issued for this by the C# compiler (CS8019), but pragmas appear to be ineffective at disabling it.  @sharwell may know why.

Reply:
> CS8019

This is a hidden diagnostic used for fading out unnecessary usings in the IDE. It is intentionally reported even in generated files, and will not impact builds.

> Do we need a new pragma warnings in the headers though?

As long as you use a hint name ending in **.g.cs** or have `// <auto-generated/>` as the first line of the file, it will be treated as generated code and code style analyzers will not run in the file.

Reply:
Thanks.

> It is intentionally reported even in generated files

I was surprised that they were still issued even with:
```C#
#pragma warning disable CS8019
```
at the top of the file.  Is that expected?

> As long as you use a hint name ending in .g.cs or have // <auto-generated/> as the first line of the file

We do both.